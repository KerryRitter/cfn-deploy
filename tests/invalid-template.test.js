const AWS = require('aws-sdk-mock');
const lib = require('../');


describe('CloudFormation.validateTemplate', () => {
  beforeAll(() => {
    AWS.mock('CloudFormation', 'validateTemplate', (params, callback) => {
      callback(new Error('INVALID_TEMPLATE'));
    });
  });

  it('should error on invalid template', (done) => {
    lib({
      region: 'us-east-1',
      stackName: 'test-stack',
      template: './tests/templates/invalid-template.yaml',
    }).on('ERROR', (err) => {
      expect(err.message).toMatchSnapshot();
      done();
    });
  });

  afterAll(() => {
    AWS.restore('CloudFormation');
  });
});
