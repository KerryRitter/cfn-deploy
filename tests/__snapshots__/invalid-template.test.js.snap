// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CloudFormation.validateTemplate should error on invalid template 1`] = `"INVALID_TEMPLATE"`;
